packages/shared/src/hooks/post/useMutateComment.ts

```diff
@@ -126,6 +126,23 @@ export const useMutateComment = ({
         children: parent.node.children,
       };
       return copy;
+    };
+
+    const forInvalidation = [];
+
+    getAllCommentsQuery(postId).forEach((queryKey) => {
+      client.setQueryData<PostCommentsData>(queryKey, (data) => {
+        if (!data) {
+          forInvalidation.push(queryKey);
+          return null;
+        }
+
+        return updateQueryData(data);
+      });
+    });
+
+    forInvalidation.forEach(async (queryKey) => {
+      await client.invalidateQueries(queryKey);
```

Comment:
This will just conflict with the read replica on this call though.
I validated it today as it was only using the optimistic update.
With invalidation we can't guarantee its available unless we remove read replica for this.

I still think dealing with it optimistic might be better?
Cc @capJavert as it impacts api

Reply:
Good point. I think we can actually even omit the invalidation of queries. The case I am worried about is that:
- User opens a post with many comments.
- Made a comment.
- This pushes a comment to the cache on all keys.
- The key for the currently displayed data is correct since it has the previous comments data.
- The other key without pre-existing data, will now have some data as we will push it, but then it would be missing all of the other comments data since this is now considered as valid (not stale) data.

Let me update it one more time.

Reply:
I think core issue is that we are trying to invalidate every key we know could exist vs invalidating what is active on the screen. 

But to focus on problem on hand, in regards to read replica we should always focus to optimistic update current view/screen/component and everything else theoretically could be just invalidated because until user navigates to it replication should be done.

To summarize:
- optimistic update comment list user is currently on
- invalidate other lists (with top comment/post key)

Reply:
Right. Then we should be good here if that is the case.